### PR TITLE
WIP: Refactor resources to to fulfill index and show

### PR DESCRIPTION
**Pull request by bambamboole**
This PR aims to solve that currently the Resources are only for Forms. It implements the logic to retrieve lists of Resources and display dynamic table in the frontend as well as dynamic show pages.
Now we have implement 3 components for each field. Index, Details and Form.

- [x] New `ResourcesIndex` Vue component
- [x] New `ResourceDetails` Vue component 
- [x]  Implemented a basic set of index and detail fields
- [x] Refactored all the logic from a Form class to a Resource class.
- [x]  New controllers for the index and show handling

### CODE REVIEW

File: resources/js/components/ResourceForm.vue [hunk 1]
```diff
@@ -125,15 +125,14 @@
...
-            prepareParams() {
-                // Filter null values. This way we can handle create and update
-                return [this.resource, this.resourceId].filter(el => el !== null)
-            },
-
             async fetchResourceForm() {
                 this.isLoading = true;
                 // fetch the form definition from the backend.
-                const response = await api(Cms.route('cms.backend.forms.show', this.prepareParams()));
+                console.log(this.resourceId)
```
Comment (slashplusweb) [inline resources/js/components/ResourceForm.vue:131]:
```suggestion

```

File: resources/js/components/ResourceIndex.vue [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                <td class="px-6 py-4 whitespace-no-wrap border-b border-gray-200">
+                    <router-link v-if="item.displayShowButtonOnIndex"
+                                 :to="{name:`${resource}s.show`, params: {id: item.resourceId}}" class="btn">
+                        show
+                    </router-link>
+                    <router-link v-if="item.displayEditButtonOnIndex && meta.is_editable"
+                                 :to="{name:`${resource}s.edit`, params: {id: item.resourceId}}"
+                                 class="btn">
+                        edit
+                    </router-link>
```
Comment (slashplusweb) [inline resources/js/components/ResourceIndex.vue:36]:
```suggestion
                    <RouterLink 
                        v-if="item.displayShowButtonOnIndex"
                        :to="{name:`${resource}s.show`, params: {id: item.resourceId}}" 
                        class="btn"
                     >
                        show
                    </RouterLink>
                    <RouterLink 
                        v-if="item.displayEditButtonOnIndex && meta.is_editable"
                       :to="{name:`${resource}s.edit`, params: {id: item.resourceId}}"
                       class="btn"
                    >
                        edit
                    </RouterLink>
```

File: resources/js/components/index/ImageField.vue [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<template>
+        <img class="rounded w-20 h-20"
+             :class="{'rounded-full': field.rounded}" :src="field.value" alt=""/>
```
Comment (slashplusweb) [inline resources/js/components/index/ImageField.vue:3]: if you start to wrap please wrap all lines.

Comment (bambamboole) [reply resources/js/components/index/ImageField.vue:3]: yes sir!

**Comment by bambamboole** [comment]
Now the Resource expects an Eloquent model. In the first instance I thought it would be nice to be independent of Eloquent, but it gives us so many advantages such as pagination, and relation and so on.

**Pull request merged**